src/main/java/com/commercetools/sync/producttypes/utils/ProductTypeUpdateEnumActionsUtils.java

```diff
@@ -0,0 +1,236 @@
+     * @param oldEnumValues           the list of old enum values.
+     * @param newEnumValues           the list of new enum values.
+     * @param <T>                     the enum type of the elements of the list.
+     * @return a list of enum values update actions if there are old enum value
+     *         that should be removed.
+     *         Otherwise, if the enum values are identical, an empty optional is returned.
+     */
+    @Nonnull
+    public static <T extends WithKey> Optional<UpdateAction<ProductType>> buildRemoveEnumValuesUpdateActions(
+        @Nonnull final String attributeDefinitionName,
+        @Nonnull final List<T> oldEnumValues,
+        @Nullable final List<T> newEnumValues) {
+
+        final List<String> keysToRemove;
+
+        if (newEnumValues != null && !newEnumValues.isEmpty()) {
+            final Map<String, T> newEnumValuesKeyMap = getEnumValuesKeyMapWithKeyValidation(
+                attributeDefinitionName,
+                newEnumValues
+            );
+
+            keysToRemove = oldEnumValues
+                .stream()
+                .map(WithKey::getKey)
+                .filter(oldEnumValueKey -> !newEnumValuesKeyMap.containsKey(oldEnumValueKey))
+                .collect(Collectors.toList());
+        } else {
+            keysToRemove = oldEnumValues
+                .stream()
+                .map(WithKey::getKey)
+                .collect(Collectors.toList());
+        }
+
+        if (!keysToRemove.isEmpty()) {
+            return ofNullable(RemoveEnumValues.of(attributeDefinitionName, keysToRemove));
+        } else {
+            return empty();
+        }
+    }
+
+    /**
+     * Compares the order of a list of old enum values and a list of new enum values. If there is a change in order,
+     * then a change of enum values order (with the new order) is built. If there are no changes in order an empty
+     * optional is returned.
+     *
+     * @param attributeDefinitionName the attribute definition name whose enum values belong to.
+     * @param oldEnumValues           the list of old enum values.
+     * @param newEnumValues           the list of new enum values.
+     * @param changeOrderEnumCallback the function that is called to apply the change in the order.
+     * @param <T>                     the enum type of the elements to change the order for.
+     * @return an optional update action if the the order of the enum values is not identical.
+     *         Otherwise, if the enum values order is identical, an empty optional is returned.
+     */
+    @Nonnull
+    public static <T extends WithKey> Optional<UpdateAction<ProductType>> buildChangeEnumValuesOrderUpdateAction(
+        @Nonnull final String attributeDefinitionName,
+        @Nonnull final List<T> oldEnumValues,
+        @Nonnull final List<T> newEnumValues,
+        @Nonnull final BiFunction<String, List<T>, UpdateAction<ProductType>> changeOrderEnumCallback) {
```

Comment:
It has to be the parent class because the action can be `ChangeEnumValueOrder` or `ChangeLocalizedEnumValueOrder`
 depending on the `changeOrderEnumCallback` will return one or the other

Another thing is that I can treat all update action as a more generic `UpdateAction<ProductType>` type, so then all update actions applied to the product type are the same and easier to manipulate in more abstract methods used at higher levels of the product type sync